src/Proxy/UserContextSubscriber.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/**
+ * File containing the UserContextSubscriber class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\PlatformHttpCacheBundle\Proxy;
+
+use FOS\HttpCache\SymfonyCache\UserContextSubscriber as BaseUserContextSubscriber;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Extends UserContextSubscriber from FOSHttpCache to include original request.
+ *
+ * {@inheritdoc}
+ */
+class UserContextSubscriber extends BaseUserContextSubscriber
+{
+    protected function cleanupHashLookupRequest(Request $hashLookupRequest, Request $originalRequest)
+    {
+        parent::cleanupHashLookupRequest($hashLookupRequest, $originalRequest);
+        // Embed the original request as we need it to match the SiteAccess.
+        $hashLookupRequest->attributes->set('_ez_original_request', $originalRequest);
```

Comment:
Is this still relevant @bdunogier / @lolautruche ? Ref:
https://github.com/ezsystems/ezpublish-kernel/blob/7.0/eZ/Publish/Core/MVC/Symfony/EventListener/SiteAccessMatchListener.php#L71-L74

Reply:
It is indeed very contradictory. I don't have a good answer now. The only thing left to do is I think annotate the files, and see what was added when, and for what reason. Unless we're very unlucky, we should be able to make sense out of it :)

Reply:
Yes it's still relevant AFAICS. The problem is probably the obsolete comment.
Comes from https://github.com/ezsystems/ezpublish-kernel/commit/880dc52d3c680f4e6ff5cdd1e559babda48a264b to fix https://jira.ez.no/browse/EZP-23632